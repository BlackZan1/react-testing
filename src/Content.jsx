import { useState } from 'react'

// assets
import logo from './logo.svg'

export const Content = () => {
    const [clicked, setClicked] = useState(false)
    const [value, setValue] = useState('')

    return (
        <div className='App'>
            <header className='App-header'>
                <img src={logo} className='App-logo' alt='logo' />

                <p>
                    Edit <code>src/App.js</code> and save to reload.
                </p>

                <button 
                    data-testid='app-btn' 
                    onClick={() => setClicked(!clicked)}
                >
                    Click me
                </button>

                {
                    clicked && (
                    <a
                        className='App-link'
                        href='https://reactjs.org'
                        target='_blank'
                        rel='noopener noreferrer'
                    >
                        Learn React
                    </a>
                    )
                }
            </header>

            <div>
                <h1>
                    <code data-testid='app-value'>
                    { value }
                    </code>
                </h1>

                <input 
                    data-testid='app-input' 
                    onChange={(ev) => setValue(ev.currentTarget.value)}
                />
            </div>      
        </div>
    )
}