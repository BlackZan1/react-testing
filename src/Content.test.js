import { fireEvent, render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Content } from './Content'

describe('Content component', () => {
  test('Must render link learn react after clicking on button', async () => {
    render(
      <Content />
    )

    expect(screen.queryByText(/learn react/i)).toBeNull()

    const btn = await screen.findByTestId('app-btn')

    fireEvent.click(btn)
  
    expect(screen.queryByText(/learn react/i)).toBeInTheDocument()

    fireEvent.click(btn)

    expect(screen.queryByText(/learn react/i)).toBeNull()
  })

  test('Must fill element by text after changing input', async () => {
    render(
      <Content />
    )

    const input = await screen.findByTestId('app-input')

    expect(screen.queryByTestId('app-value')).toContainHTML('')

    // fireEvent.change(input, {
    //   target: {
    //     value: '123123'
    //   }
    // })

    userEvent.type(input, '123123')

    expect(screen.queryByTestId('app-value')).toContainHTML('123123')
    expect(screen.queryByTestId('app-value')).toMatchSnapshot()
  })
})
