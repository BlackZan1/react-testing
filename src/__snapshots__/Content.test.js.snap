// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content component Must fill element by text after changing input 1`] = `
<code
  data-testid="app-value"
>
  123123
</code>
`;
